orttraining/orttraining/test/graph/module_gradient_graph_builder_test.cc

```diff
@@ -0,0 +1,183 @@
+                                       split_graphs_info.backward_intializer_names_as_input.begin(),
+                                       split_graphs_info.backward_intializer_names_as_input.end());
+  expected_backward_input_names.insert(expected_backward_input_names.end(),
+                                       split_graphs_info.intermediate_tensor_names.begin(),
+                                       split_graphs_info.intermediate_tensor_names.end());
+  expected_backward_input_names.insert(expected_backward_input_names.end(),
+                                       split_graphs_info.backward_output_grad_names.begin(),
+                                       split_graphs_info.backward_output_grad_names.end());
+  expected_backward_output_names.insert(expected_backward_output_names.end(),
+                                        split_graphs_info.initializer_grad_names_to_train.begin(),
+                                        split_graphs_info.initializer_grad_names_to_train.end());
+
+  const std::vector<const NodeArg*>& forward_graph_inputs = forward_model.MainGraph().GetInputsIncludingInitializers();
+  for (auto& node_arg : forward_graph_inputs) {
+    actual_forward_input_names.emplace_back(node_arg->Name());
+  }
+
+  const std::vector<const NodeArg*>& forward_graph_outputs = forward_model.MainGraph().GetOutputs();
+  for (auto& node_arg : forward_graph_outputs) {
+    actual_forward_output_names.emplace_back(node_arg->Name());
+  }
+
+  const std::vector<const NodeArg*>& backward_graph_inputs =
+      backward_model.MainGraph().GetInputsIncludingInitializers();
+  for (auto& node_arg : backward_graph_inputs) {
+    actual_backward_input_names.emplace_back(node_arg->Name());
+  }
+
+  const std::vector<const NodeArg*>& backward_graph_outputs = backward_model.MainGraph().GetOutputs();
+  for (auto& node_arg : backward_graph_outputs) {
+    actual_backward_output_names.emplace_back(node_arg->Name());
+  }
+
+  ASSERT_TRUE(expected_forward_input_names == actual_forward_input_names);
+  ASSERT_TRUE(expected_forward_output_names == actual_forward_output_names);
+  ASSERT_TRUE(expected_backward_input_names == actual_backward_input_names);
+  ASSERT_TRUE(expected_backward_output_names == actual_backward_output_names);
+}
+
+TEST(ModuleGradientGraphBuilderTest, GraphSplit_Mnist) {
+  std::string file_path = "testdata/test_training_model.onnx";
+  std::vector<std::string> initializer_names_to_train{"W1", "B1", "W2", "B2", "W3", "B3"};
+  std::map<std::string, int> expected_forward_ops_count = {{"Add", 3}, {"MatMul", 3}, {"Relu", 2}};
+  std::map<std::string, int> expected_backward_ops_count = {
+      {"Gemm", 5}, {"Identity", 3}, {"ReduceSum", 3}, {"com.microsoft.ReluGrad", 2}};
+
+  RunModuleGradientGraphBuilderTest(file_path, initializer_names_to_train, expected_forward_ops_count,
+                                    expected_backward_ops_count);
+}
+
+TEST(ModuleGradientGraphBuilderTest, GraphSplit_BertToy) {
```

Comment:
There are two other thing we should check. 

1) the new intermediate outputs in the fw graph
2) both the fw initializers and intermediate tensors are now listed as graph inputs in the bw graph